modules/behavioural_qc/php/endpoints/behavioural.class.inc

```diff
@@ -0,0 +1,288 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used for populating the behavioural_qc.
+ *
+ * PHP Version 7
+ *
+ * @category Behavioural
+ * @package  Loris
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\behavioural_qc\endpoints;
+
+use LORIS\behavioural_qc\Provisioners\BehaviouralProvisioner;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\api\Endpoint;
+use \LORIS\Middleware\ETagCalculator;
+
+/**
+ * The NDB_Filter_Form_DataTeamHelper Class.
+ *
+ * This class provides the php code to create the menu filter form that
+ * allows profiles to be processed as part of the genomic browser.
+ *
+ * @category Behavioural
+ * @package  Loris
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Behavioural extends Endpoint implements ETagCalculator
+{
+    /**
+     * A cache of the endpoint results, so that it doesn't need to be
+     * recalculated for the ETag and handler.
+     */
+    private $_cache;
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
```

Comment:
I'm still not sure why this is used. The module has a `hasAccess` method which should be called when the user loads the page.

Reply:
@johnsaigle you're right it doesn't make sense. I'll remove it because the hasAccess should make it so this code doesn't access when the permissions fail.